src/validation.cpp

```diff
@@ -2965,6 +2965,18 @@ bool CChainState::ActivateBestChainStep(CValidationState& state, const CChainPar
                         InvalidChainFound(vpindexToConnect.front());
                     }
                     state = CValidationState();
+                    if (pindexConnect == pindexMostWork) {
```

Comment:
Can I ask what's this checking for?

Reply:
Can we trigger this off checkpoint not matching instead or is this an easier way to check

Reply:
Just try to do it once, when we are at most work.